src/app/wishlists/wishlist-fetch.ts

```diff
@@ -0,0 +1,76 @@
+import { toWishList } from './wishlist-file';
+import { t } from 'app/i18next-t';
+import _ from 'lodash';
+import { showNotification } from 'app/notifications/notifications';
+import { loadWishLists } from './actions';
+import store from 'app/store/store';
+import { ThunkResult, RootState } from 'app/store/reducers';
+import { ThunkDispatch } from 'redux-thunk';
+import { AnyAction } from 'redux';
+
+export function fetchWishList(showAlert: boolean): ThunkResult<Promise<void>> {
+  return async (dispatch, getState) => {
+    const wishListSource = getState().settings.wishListSource;
+
+    if (!wishListSource) {
+      return;
+    }
+
+    fetch(wishListSource)
```

Comment:
Since you have an async function, you can go ahead and use async/await syntax in the whole thing.